### PR TITLE
SOV-438: trezor module

### PR DESCRIPTION
**Pull request by creed-victor**
https://sovryn.atlassian.net/browse/SOV-438

### CODE REVIEW

File: .changeset/gentle-berries-pretend.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+"web": patch
+"@sovryn/onboard-core": patch
+"@sovryn/onboard-hw-common": patch
+"@sovryn/onboard-ledger": patch
+"@sovryn/onboard-react": patch
+"@sovryn/onboard-trezor": patch
+---
+
+SOV-438: added trezor module
+also these fixes:
```
Comment (pietro-maximoff) [inline .changeset/gentle-berries-pretend.md:11]:
```suggestion
also these fixes:

```

Comment (pietro-maximoff) [reply .changeset/gentle-berries-pretend.md:11]: needs for prettier format

**Review by pietro-maximoff — COMMENTED** [review]
tested all wallets (Ledger, Trezor, MM) separately and all together at the same time. also tried to test using different derivation paths with different chain ids.  No issues were found, all works well as expected, and is build - without errors.
@creed-victor One question that I have is that when we choose the custom derivation path - we have to show the input field to be able to add a needed value, as I understand it will be done in a separate task, right?

File: .changeset/gentle-berries-pretend.md (hunk 1 above)
Comment (creed-victor) [reply .changeset/gentle-berries-pretend.md:11]: it's actually auto generated files and will be automatically deleted from git once package is published (it's used by bot/cli to prepare changelogs and bump versions correctly)

**Comment by creed-victor** [comment]
> @creed-victor One question that I have is that when we choose the custom derivation path - we have to show the input field to be able to add a needed value, as I understand it will be done in a separate task, right?

@pietro-maximoff, oh, it was there before but I hid it by mistake while adding dropdowns. Will fix it

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 2]
```diff
@@ -58,11 +80,37 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
...
+            {path.label} ({path.value})
+          </option>
+        ))}
+        <option value="custom">Custom</option>
+      </select>
+
+      {derivationPathSelect === "custom" && (
+        <input
+          value={customPath}
+          onChange={(e) => setCustomPath(e.target.value)}
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:95]: Use `useCallback`.

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 3]
```diff
@@ -58,11 +80,37 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
...
+        />
+      )}
+
+      <select onChange={(e) => setChainId(e.target.value)}>
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:99]: Use `useCallback`.

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 4]
```diff
@@ -58,11 +80,37 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
...
+        {selectAccountOptions.chains.map((chain) => (
+          <option value={chain.id} key={chain.id}>
+            {chain.label}
+          </option>
+        ))}
+      </select>
+
+      <select onChange={(e) => setAsset({ label: e.target.value })}>
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:107]: Use `useCallback`.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 5]
```diff
@@ -14,14 +15,24 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
   const [scanning, setScanning] = useState(false);
   const [error, setError] = useState<string>();
 
-  const [derivationPath, setDerivationPath] = useState<string>(
+  const [derivationPathSelect, setDerivationPath] = useState<string>(
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:18]:
```suggestion
  const [derivationPathSelect, setDerivationPathSelect] = useState(
```

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 6]
```diff
@@ -14,14 +15,24 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
...
     selectAccountOptions.basePaths[0].value
   );
   const [chainId, setChainId] = useState<string>(
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:21]:
```suggestion
  const [chainId, setChainId] = useState(
```

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 7]
```diff
@@ -14,14 +15,24 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
...
     selectAccountOptions.chains[0].id
   );
+  const [customPath, setCustomPath] = useState<string>(
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:24]:
```suggestion
  const [customPath, setCustomPath] = useState(
```

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 8]
```diff
@@ -14,14 +15,24 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
...
+    selectAccountOptions.basePaths[0].value
+  );
   const [asset, setAsset] = useState<Asset>(selectAccountOptions.assets[0]);
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:27]:
```suggestion
  const [asset, setAsset] = useState(selectAccountOptions.assets[0]);
```

File: packages/onboard-react/src/SelectAccount/index.tsx [hunk 9]
```diff
@@ -45,7 +56,39 @@ export const SelectAccount: FC<SelectAccountProps> = () => {
...
+    (event: ChangeEvent<HTMLSelectElement>) => {
+      setAsset({ label: event.target.value });
+    },
+    []
+  );
+
+  const handleAccountSelect = useCallback(
     (account: Account) => () => {
       selectAccount(account);
     },
```
Comment (tiltom) [inline packages/onboard-react/src/SelectAccount/index.tsx:94]:
```suggestion
  const handleCustomPathChange = useCallback(
    (event: ChangeEvent<HTMLInputElement>) => setCustomPath(event.target.value),
    []
  );

  const handleChainChange = useCallback(
    (event: ChangeEvent<HTMLSelectElement>) => setChainId(event.target.value),
    []
  );

  const handleAssetChange = useCallback(
    (event: ChangeEvent<HTMLSelectElement>) => setAsset({ label: event.target.value }),
    []
  );

  const handleAccountSelect = useCallback(
    (account: Account) => () => selectAccount(account),
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Comment by tiltom** [comment]
When connected with Trezor via a custom path, signing of a tx fails with a checksum error.
Screenshot 2022-09-19 at 11 37 27

**Comment by creed-victor** [comment]
> When connected with Trezor via a custom path, signing of a tx fails with a checksum error. 

fixed, should work now

**Comment by pietro-maximoff** [comment]
> > When connected with Trezor via a custom path, signing of a tx fails with a checksum error.
> 
> fixed, should work now

working for me

**Review by pietro-maximoff — APPROVED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**